pkg/logging/logging.go

```diff
@@ -40,7 +43,8 @@ const (
 
 type Driver interface {
 	Init(dataStore, ns, id string) error
-	Process(dataStore string, config *logging.Config) error
+	PreProcess(dataStore string, config *logging.Config) error
+	Process(stdout chan string, stderr chan string) error
```

Comment:
Casting params to read-only directional channels may be better IMO.
```go
Process(stdout <-chan string, stderr <-chan string) error
```

Reply:
nice catch, done